arangod/IResearch/IResearchFeature.cpp

```diff
@@ -697,31 +687,29 @@ void registerUpgradeTasks(application_features::ApplicationServer& server) {
     // will just do nothing but flags don`t allow to distinguih cases)
     task.clusterFlags = methods::Upgrade::Flags::CLUSTER_DB_SERVER_LOCAL |
                         methods::Upgrade::Flags::CLUSTER_LOCAL;  // db-server
-    task.databaseFlags =
-        methods::Upgrade::Flags::DATABASE_EXISTING |
-        methods::Upgrade::Flags::DATABASE_ONLY_ONCE;  // seal the task after
-                                                      // execution
+    task.databaseFlags = methods::Upgrade::Flags::DATABASE_EXISTING |
+                         // seal the task after execution
+                         methods::Upgrade::Flags::DATABASE_ONLY_ONCE;
     task.action = &upgradeArangoSearchLinkCollectionName;
     upgrade.addTask(std::move(task));
   }
 }
 
 void registerViewFactory(application_features::ApplicationServer& server) {
-  auto& viewType = arangodb::iresearch::DATA_SOURCE_TYPE;
-  auto& viewTypes = server.getFeature<ViewTypesFeature>();
+  static_assert(IResearchView::typeInfo() ==
+                IResearchViewCoordinator::typeInfo());
+  constexpr std::string_view kViewType{IResearchView::typeInfo().second};
 
   Result res;
 
   // DB server in custer or single-server
-  if (ServerState::instance()->isCoordinator()) {
-    res = viewTypes.emplace(
-        viewType, arangodb::iresearch::IResearchViewCoordinator::factory());
+  if (auto& viewTypes = server.getFeature<ViewTypesFeature>();
+      ServerState::instance()->isCoordinator()) {
+    res = viewTypes.emplace(kViewType, IResearchViewCoordinator::factory());
   } else if (ServerState::instance()->isDBServer()) {
-    res = viewTypes.emplace(viewType,
-                            arangodb::iresearch::IResearchView::factory());
+    res = viewTypes.emplace(kViewType, IResearchView::factory());
   } else if (ServerState::instance()->isSingleServer()) {
```

Comment:
Minor: this `else if` and the following `else if` seem to execute the same code and could be combined.